src/async_impl/client.rs

```diff
@@ -325,6 +330,12 @@ impl ClientBuilder {
     pub fn dns_threads(self, _threads: usize) -> ClientBuilder {
         self
     }
+
+    /// Bind to a local IP Address
+    pub fn local_address(mut self, addr: Option<IpAddr>) -> ClientBuilder {
```

Comment:
Consider an argument of `addr: impl Into<Option<IpAddr>>`. This way, in the common case, it reads more fluently, like `builder.local_address(ip)` instead of `builder.local_address(Some(ip))`.